ha/cli/exec/clusterExecutor.py

```diff
@@ -129,50 +136,156 @@ def execute(self):
+        functinality and which is responsible for delegating that request
+        to Cluster Manager
+    '''
 
-    def execute(self) -> str:
-        raise HAUnimplemented("This operation is not implemented.")
+    def __init__(self):
+        '''Init Method'''
+        super(ClusterNodeAddExecutor, self).__init__()
+        self._pcs_cluster_controller = PcsClusterController()
+        self._op = Output()
+        self._args = None
+
+    def parse_cluster_args(self) -> bool:
+        '''
+           Parses the command line args.
+           Return: argparse
+        '''
+        parser = argparse.ArgumentParser(prog='cluster add node')
+        parser.add_argument("cluster", help="Module")
+        parser.add_argument("add", help="action to be performed")
+        parser.add_argument("node", help="component on which action to be performed")
+        group = parser.add_mutually_exclusive_group(required='True')
+        group.add_argument('--nodeid', action='store', \
+                            help='ID of a node which needs to be added in a cluster')
+        group.add_argument('--descfile', action='store', \
+                            help='A file which describes the node to be added in a cluster', \
+                            type=self._is_file_extension_valid)
+        parser.add_argument('--username', help='cluster username', required=True)
+        parser.add_argument('--password', help='cluster password', required=True)
+        parser.add_argument('--json', help='Required output fomat', action='store_true')
+        self._args = parser.parse_args()
+        return True
+
+    def _is_file_extension_valid(self, filename) -> str:
+        '''
+           Checks if file extension which is passed is correct or not
+           Returns: str
+           Exception: ArgumentTypeError
+        '''
+        base, ext = os.path.splitext(filename)
+        if ext.lower() not in ('.json'):
+            raise argparse.ArgumentTypeError('File must have a json extension')
+        return filename
+
+    def validate(self) -> bool:
+        '''
+           Validates permission and command line arguments.
+           Exception: HAInvalidPermission
+           Return: bool
+        '''
+        # Every CLI command will be an internal command now. So,
+        # we do not need this change. If required, can be revisited later.
+        #if not self.is_ha_user():
+        #    raise HAInvalidPermission('Not enough permissions to invoke this command')
+        if self.parse_cluster_args():
+            return True
+        return False
+
+    def _is_valid_node_id(self, node_id) -> bool:
+        '''
+           Checks if node id gets resolved to some IP address or not
+           Returns: bool
+           Exception: socket.gaierror, socket.herror
+        '''
+        ip_validator_regex = "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$"
```

Comment:
I think we were supposed just to try pinging whatever node_id we get here, why are we doing more than that here?

Reply:
Because, that node_id validation using system health list is not there yet. So, anyone can execute this command with any value of node_id. I tried that way earlier and it broke. so, I kept it this way. I can remove it if needed. Please let me know

Reply:
ok